engines/order_management/app/services/order_management/order/updater.rb

```diff
@@ -35,6 +35,14 @@ def update_totals_and_states
         end
 
         persist_totals
+        update_pending_payment
+      end
+
+      def update_pending_payment
+        return unless order.state.in? ["payment", "confirmation"]
+        return unless order.pending_payments.any?
+
+        order.pending_payments.first.update_attribute :amount, order.total
```

Comment:
Yeah, I haven't dug into this here, but I wanted to. It's the same logic that's been moved from that callback, but it does feel a bit simplistic. Are there cases where there could be multiple pending payments, and if so wouldn't `last` be more sensible then `first` here, as it would be the most recent? :thinking:

Reply:
We do have a couple of mechanisms in place for making sure there's only one payment on a pre-completed order during checkout. Maybe some exotic tests that try to intentionally create edge cases might be valuable here..?

Reply:
I'm happy if we have only one pending payment at a time. It would just be good to report it if that's somehow not the case.

Reply:
Out of scope here though. I saw that you didn't change this code.